### PR TITLE
`RecFreeMap` in `Filter`

### PR DESCRIPTION
**Pull request by jsantos17**
#qz-3777 done

### CODE REVIEW

**Comment by alissapajer** [comment]
There will probably be merge conflicts in the imports between this and #3560.

**Comment by jsantos17** [comment]
@alissapajer I'll go through the conflicts and fix them. There are a lot, but the changes in this PR are fairly mechanical and fixing the conflicts should be fairly easy to do

File: connector/src/main/scala/quasar/qscript/rewrites/Normalizable.scala [hunk 1]
```diff
@@ -228,9 +229,9 @@ class NormalizableT[T[_[_]]: BirecursiveT: EqualT: ShowT: RenderTreeT] extends T
         makeNorm(bucket, order)(rebucket, _ => orderNormOpt)(Sort(src, _, _))
 
       case Map(src, f)                   => freeMFEq(f.linearize).map(RecFreeS.fromFree(_)).map(Map(src, _))
-      case LeftShift(src, s, i, t, u, r) => makeNorm(s.linearize, r)(freeMFEq(_), freeMFEq(_))((str, repair) => LeftShift(src, RecFreeS.fromFree(str), i, t, u, repair))
+      case LeftShift(src, s, i, t, u, r) => makeNorm(s.linearize, r)(freeMFEq(_), freeMFEq(_))((str, repair) => LeftShift(src, str.asRec, i, t, u, repair))
       case Union(src, l, r)              => makeNorm(l, r)(freeTCEq(_), freeTCEq(_))(Union(src, _, _))
-      case Filter(src, f)                => freeMFEq(f).map(Filter(src, _))
+      case Filter(src, f)                => freeMFEq(f.linearize).map(fm => Filter(src, fm.asRec))
```
Comment (djspiewak) [inline connector/src/main/scala/quasar/qscript/rewrites/Normalizable.scala]: As with everything else, I assume we're going to come back to these sorts of things later to ensure that we preserve let bindings through normalizations?

Comment (jsantos17) [reply connector/src/main/scala/quasar/qscript/rewrites/Normalizable.scala]: Yep. After this PR I'll attempt to create some let-bindings for the structure editor queries and will probably need to fix the codepaths where the let bindings are `linearize`d to get it working.

Comment (jsantos17) [reply connector/src/main/scala/quasar/qscript/rewrites/Normalizable.scala]: @djspiewak In the case of normalization on `FreeMap`s like this one, I've been thinking these can be safely applied to `RecFreeMap` by applying them to the bound form and body separately.

**Pull request merged**